### PR TITLE
initial checkin for vehicle, customer and contract

### PR DESCRIPTION
**Pull request by cheeyim**

### CODE REVIEW

**Comment by santiagobasulto** [comment]
This PR will be reviewed by: @brianr001

File: dealership/vehicles.py [hunk 1]
```diff
@@ -1,15 +1,51 @@
...
+        
+    def purchase_price(self):
+        raise NotImplementedError()
 
 class Car(Vehicle):
-    pass
+    
+    INTEREST = 1.07
+    SALE_MULTIPLIER = 1.2
+    PURCHASE_MULTIPLIER = 0.004
```
Comment (brianr001) [inline dealership/vehicles.py]: Compared to my solution, I think it makes more sense to do as you did and put the Interest and multipliers in the classes for the individual types of vehicles. Smart move.

File: dealership/vehicles.py [hunk 2]
```diff
@@ -1,15 +1,51 @@
...
+        
+    def purchase_price(self):
+        return self.sale_price() - (self.PURCHASE_MULTIPLIER * self.miles)
 
 
 class Truck(Vehicle):
-    pass
+    
+    INTEREST = 1.11
+    SALE_MULTIPLIER = 1.7
```
Comment (brianr001) [inline dealership/vehicles.py]: This was confusing me, as I think that what you have listed as the SALE_MULTIPLIER is actually the LEASE_MULTIPLIER? It looks like you hard-coded the actual SALE-MULTIPLIER (1.6) below.

File: dealership/contracts.py [hunk 3]
```diff
@@ -1,12 +1,46 @@
...
 
 class BuyContract(Contract):
     def __init__(self, vehicle, customer, monthly_payments):
-        pass
-
+        super(BuyContract, self).__init__(vehicle, customer)
+        self.monthly_payments = monthly_payments
+        
+    def total_value(self):
+
```
Comment (brianr001) [inline dealership/contracts.py]: You've got some weird spacing choices (just like me!) inside your methods. You could probably clean those up a bit to improve readability.

Very nice job, Chee Yim!

File: dealership/vehicles.py (hunk 2 above)
Comment (cheeyim) [reply dealership/vehicles.py]: Good catch! Yes, it should be LEASE_MULTIPLIER. Updated it

File: dealership/contracts.py (hunk 3 above)
Comment (cheeyim) [reply dealership/contracts.py]: Got it! Learnt that I need to add backslash to write codes in multiple lines (if we want to break up lines that are too long). Also clean up the spaces.. 

Thanks for the constructive feedback @brianr001 !

**Pull request closed without merging**